types/next/document.d.ts

```diff
@@ -104,5 +104,5 @@ export class NextScript extends React.Component<NextScriptProps> {}
 export default class Document<P = {}> extends React.Component<
     P & DefaultDocumentIProps & DocumentProps
 > {
-    static getInitialProps(context: NextDocumentContext): DefaultDocumentIProps;
+    static getInitialProps(context: NextDocumentContext): DefaultDocumentIProps | Promise<DefaultDocumentIProps>;
```

Comment:
Shouldn't we just make this return the `Promise` instead? @joaovieira

Reply:
No, it doesn't need to be a Promise necessarily (e.g. the default Document is not). I made these to map the exact signature of the exported classes, but again it needs some visiting if sub-classes have to adhere to the same signature.